### PR TITLE
conf: comment on the status-addr to notice the risk

### LINKED ISSUE #7006: Comment on the status server port to notify the potential information leaks
### Is your feature request related to a problem? Please describe:

TiKV exports a status port to supply the metrics or memory profile. The users have a risk of leaking sensitive information if they expose this port to the public by mistake.

### Describe the feature you'd like:

Comment the risk in the configuration file.

### PR DESCRIPTION
**Pull request by shafreeck**
Signed-off-by: Shafreeck Sea <shafreeck@gmail.com>

fixes #7006 

###  What have you changed?

Update the comment of the configuration template to notice the risk to expose the status server port to the public.

###  What is the type of changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- No code

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

NO

###  Does this PR affect `tidb-ansible`?

NO

###  Refer to a related PR or issue link (optional)

#7006

### CODE REVIEW

File: etc/config-template.toml [hunk 1]
```diff
@@ -105,7 +105,9 @@
 # advertise-addr = ""
 
 ## Status address.
-## This is used for reporting the status of TiKV directly through the HTTP address.
+## This is used for reporting the status of TiKV directly through 
+## the HTTP address. Notice that there is a risk of leaking status
+## information if this port is exposed to the public.
```
Comment (BusyJay) [inline etc/config-template.toml]: Why not support https directly?

Comment (shafreeck) [reply etc/config-template.toml]: I think TLS is another problem, it is handled by another issue #7014 .

The problem is even the TLS is supported, it needs the user to configure the client verification which means signing certificates for clients.

So TLS has two propose here:

1. Encrypt the connection which prevents the middle-man attack.
2. Verify the client which can also be solved by some simpler authentication method(like basic auth).

1  can not prevent information leaks if the port is public, 2 is hard to get enough attention to push users to verify the clients.

Comment (BusyJay) [reply etc/config-template.toml]: So to make it secure totally, we can support TLS + basic auth, right?

Comment (gregwebs) [reply etc/config-template.toml]: In a k8s deployment with tidb-operator we can automate the setup of certs or basic auth. I don't think we do basic auth elsewhere, so I don't think we should start doing it here without additional thought as to whether this is an approach we would use more uniformly.

Comment (tennix) [reply etc/config-template.toml]: We can have basic authorization based on CommonName field as suggested in https://github.com/tikv/tikv/issues/6982. tikv-server can be configured to only allow specific client that has the right CN.

Comment (gregwebs) [reply etc/config-template.toml]: Yes, this is the TLS auth approach. The CN for this status port would be different than the CN for normal data access, so it needs to be configured separately.

**Pull request merged**